dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/CosmosNoSqlCollection.cs

```diff
@@ -296,12 +296,12 @@ private async Task UpsertCoreAsync(TRecord record, int recordIndex, IReadOnlyLis
 
         if (string.IsNullOrWhiteSpace(keyValue))
         {
-            throw new VectorStoreOperationException($"Key property {this._model.KeyProperty.ModelName} is not initialized.");
+            throw new VectorStoreException($"Key property {this._model.KeyProperty.ModelName} is not initialized.");
```

Comment:
In other cases, we throw `VectorStoreException` when a service/db call has failed. Here it's clear the collection was created without required data being provided. Should we just throw `ArgumentException` during collection creation instead?

Reply:
Agree!